eclair-core/src/test/scala/fr/acinq/eclair/blockchain/DummyOnChainWallet.scala

```diff
@@ -40,8 +41,11 @@ class DummyOnChainWallet extends OnChainWallet {
 
   override def getReceivePubkey(receiveAddress: Option[String] = None)(implicit ec: ExecutionContext): Future[Crypto.PublicKey] = Future.successful(dummyReceivePubkey)
 
-  override def makeFundingTx(pubkeyScript: ByteVector, amount: Satoshi, feeRatePerKw: FeeratePerKw)(implicit ec: ExecutionContext): Future[MakeFundingTxResponse] =
-    Future.successful(DummyOnChainWallet.makeDummyFundingTx(pubkeyScript, amount))
+  override def makeFundingTx(pubkeyScript: ByteVector, amount: Satoshi, feeRatePerKw: FeeratePerKw)(implicit ec: ExecutionContext): Future[MakeFundingTxResponse] = {
+    val tx = DummyOnChainWallet.makeDummyFundingTx(pubkeyScript, amount)
+    funded += (tx.fundingTx.txid -> tx.fundingTx)
+    Future.successful(tx)
```

Comment:
I'm making many changes to these test helpers in #2275, if you plan on making more changes here in follow-up PRs we should probably do it after dual-funding to minimize conflicts.

Reply:
Gotcha, I don't plan to make more changes so we should be fine. Dealing with the mempool/blockchain is one of the trickiest parts, because they need to be shared among various `MinimalNodeFixture`. I have explored several approaches and am still not 100% happy.

In particular, there are two ways of confirming channels: `MinimalNodeFixture.confirmChannel` and ``MinimalNodeFixture.autoConfirmLocalChannels`. I think the latter (based on autopilot) is nicer when you don't need fine-grained control, but the core issue of having access to the actual transaction remains. For example, note that we pass a reference to alice's wallet for bob's autopilot:
https://github.com/ACINQ/eclair/blob/2e6e566ccc0dbaa49ed9678634b33df0d97f216a/eclair-core/src/test/scala/fr/acinq/eclair/integration/basic/TwoNodesIntegrationSpec.scala#L50

Maybe with dual-funding this will be a tad simpler since both sides know the full tx?